### PR TITLE
Tables dark mode- RSC-1043

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://issues.sonatype.org/browse/RSC-1043

-I still have to update one snapshot that includes pagination (I will wait for [RSC-1033](https://issues.sonatype.org/browse/RSC-1033) to finish), and will update 2 NxFilterDropdown snapshots located inside a table once [RSC-1564](https://issues.sonatype.org/browse/RSC-1564) is complete

-There are some discrepancies of interaction states between NxTable & NxList/NxCollapsibleItems in light mode. I have reviewed them with Jason and they asked to explore this in a separate ticket ([RSC-1648](https://issues.sonatype.org/browse/RSC-1648)), which may clean up a couple table variables in the future

### CODE REVIEW

**Comment by EmilyBermudez88** [comment]
>I still have to update one snapshot that includes pagination (I will wait for [RSC-1033](https://issues.sonatype.org/browse/RSC-1033) to finish), and will update 2 NxFilterDropdown snapshots located inside a table once [RSC-1564](https://issues.sonatype.org/browse/RSC-1564) is complete

NxTable snapshots updated that have pagination included, NxPolicyViolationIndicator snapshot with a table included, and all NxFilterDropdown snapshots. There was a mixup of which snapshots should be included in RSC-1564 (RSC https://github.com/sonatype/sonatype-react-shared-components/pull/1061#issuecomment-1514960195), and need to update all NxFilterDropdown snapshots instead of just the ones to include NxTable in them.

**Comment by cindyko226** [comment]
Should this be reviewed until the [RSC-1648](https://issues.sonatype.org/browse/RSC-1648) is done?

**Comment by EmilyBermudez88** [comment]
> Should this be reviewed until the [RSC-1648](https://issues.sonatype.org/browse/RSC-1648) is done?

Yep, this is currently on hold until RSC-1648 is dealt with.

File: gallery/visualtests/__image_snapshots__/nx-table-js-nx-table-looks-right-with-a-pagination-bar-and-a-filter-row-1.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots__/nx-table-js-nx-table-looks-right-with-a-pagination-bar-and-a-filter-row-1.png:1]: Not sure why this is in this PR

File: gallery/visualtests/__image_snapshots_dark__/nx-table-js-nx-table-looks-right-with-an-active-sort-column-1-dark.png
Comment (rpokorny) [inline gallery/visualtests/__image_snapshots_dark__/nx-table-js-nx-table-looks-right-with-an-active-sort-column-1-dark.png:1]: This test appears to be broken.  It's supposed to be showing the active color of the sort caret.  The light mode snapshot is equally broken.  Relatedly, where are the mockups for active sort carets in dark mode?

File: lib/src/base-styles/_nx-colors.scss [hunk 1]
```diff
@@ -485,6 +490,11 @@
     --nx-color-policy-threat-slider-text-disabled: var(--nx-color-policy-threat-slider-border-disabled);
     --nx-color-policy-threat-slider-text: var(--nx-swatch-indigo-97);
 
+    // NxTable
+    --nx-color-table-header-background: var(--nx-swatch-indigo-35);
+    --nx-color-table-row-background-hover: var(--nx-swatch-indigo-15);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-colors.scss:495]: This matches `NxList`, so there should probably be a common variable for at least those two.  I am however wondering why they use indigo-15 while other interactive elements (including other elements that have a `selected` state, like `NxCollapsibleItems` use indigo-10.

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx [hunk 2]
```diff
@@ -724,6 +724,21 @@ const CssVariablesPage = () => {
             <ColorDocRow colorVar="--nx-color-policy-threat-slider-text">
               The default text color of <NxCode>NxPolicyThreatSlider</NxCode>.
             </ColorDocRow>
+            <ColorDocRow colorVar="--nx-color-table-header-background">
+              The background color of the <NxCode>NxTable</NxCode> header.
+            </ColorDocRow>
+            <ColorDocRow colorVar="--nx-color-table-row-background-selected">
```
Comment (rpokorny) [inline gallery/src/styles/CssVariables/CssVariablesPage.tsx:730]: Some of these variables have been removed

File: lib/src/base-styles/_nx-tables.scss [hunk 3]
```diff
@@ -8,11 +8,7 @@
 @use '../scss-shared/nx-text-helpers';
 @use '../scss-shared/nx-button-helpers';
 @use '../scss-shared/_nx-table-variables';
-
-%table-outline {
-  outline-offset: -2px;
-  outline: 2px solid;
-}
+@use '../scss-shared/nx-dark-mode-helpers';
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-tables.scss:11]:
```suggestion
```

File: lib/src/base-styles/_nx-colors.scss (hunk 1 above)
Comment (EmilyBermudez88) [reply lib/src/base-styles/_nx-colors.scss:495]: >I am however wondering why they use indigo-15 while other interactive elements (including other elements that have a `selected` state, like `NxCollapsibleItems` use indigo-10.

I know this was brought up by Abhinav a week or two ago, and he had a meeting with Jason to confirm the interaction states and choices made: https://sonatype.slack.com/archives/CJUF9G7T2/p1681828099480709

File: gallery/visualtests/__image_snapshots_dark__/nx-table-js-nx-table-looks-right-with-an-active-sort-column-1-dark.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots_dark__/nx-table-js-nx-table-looks-right-with-an-active-sort-column-1-dark.png:1]: So Jason made a ticket to investigate this & other possible interaction states for the caret in [RSC-1646](https://issues.sonatype.org/browse/RSC-1646). They wanted to wait to update the mocks until then, but agree on the use of `purple-80` at this point

File: lib/src/base-styles/_nx-colors.scss (hunk 1 above)
Comment (EmilyBermudez88) [reply lib/src/base-styles/_nx-colors.scss:495]: > This matches NxList, so there should probably be a common variable for at least those two. I am however wondering why they use indigo-15 while other interactive elements (including other elements that have a selected state, like NxCollapsibleItems use indigo-10.

Looking through the dark mode hover variables, `indigo-15` is used on hover for secondary/tertiary buttons, dropdownMenus, tabs, and tables. (`indigo-10` used by collapsibleItems, pagination & iconOnly buttons I believe). Would a valid generic name simply be `--nx-color-interactive-background-hover-secondary`?

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx (hunk 2 above)
Comment (EmilyBermudez88) [reply gallery/src/styles/CssVariables/CssVariablesPage.tsx]: > Some of these variables have been removed

Thanks for the reminder [44bc748](https://github.com/sonatype/sonatype-react-shared-components/pull/1072/commits/44bc7482c2d9f7bb79b85aaa7e33c0ad1fd592e4)

File: gallery/visualtests/__image_snapshots__/nx-table-js-nx-table-looks-right-with-a-pagination-bar-and-a-filter-row-1.png
Comment (EmilyBermudez88) [reply gallery/visualtests/__image_snapshots__/nx-table-js-nx-table-looks-right-with-a-pagination-bar-and-a-filter-row-1.png:1]: > Not sure why this is in this PR

This came from the pagination dark mode ticket (https://github.com/sonatype/sonatype-react-shared-components/pull/1063)- light mode was missing semibold text for the selected button and was fixed there. Should I remove and leave this for final cleanup?

Comment (rpokorny) [reply gallery/visualtests/__image_snapshots__/nx-table-js-nx-table-looks-right-with-a-pagination-bar-and-a-filter-row-1.png:1]: > This came from the pagination dark mode ticket (https://github.com/sonatype/sonatype-react-shared-components/pull/1063 light mode was missing semibold text for the selected button and was fixed there. Should I remove and leave this for final cleanup?

I guess it's ok to have it here

**Review by rpokorny — APPROVED** [review]

**Pull request merged**